src/git/utils.ts

```diff
@@ -152,14 +161,30 @@ function textToPackedRefs(text: string): Refs {
   return refs;
 }
 
+/**
+ * Reads and parses the packed-refs file.
+ * @param fs
+ * @param gitdir
+ */
 async function packedRefs(fs: EncryptedFS, gitdir: string): Promise<Refs> {
-  const text = await fs.promises.readFile(path.join(gitdir, 'packed-refs'), {
-    encoding: 'utf8',
-  });
-  const refs = textToPackedRefs(text.toString());
-  return refs;
+  let text: string | Buffer = '# pack-refs with: peeled fully-peeled sorted';
+  try {
+    text = await fs.promises.readFile(path.join(gitdir, 'packed-refs'), {
+      encoding: 'utf8',
+    });
+  } catch (err) {
+    if (err.code !== 'ENOENT') throw err;
+    // If no file then ignore and return default.
+  }
+  return textToPackedRefs(text!.toString());
 }
 
+/**
+ * Obtains a list of all refs by recursivly reading the FS.
+ * @param fs
+ * @param gitdir
+ * @param filepath
+ */
```

Comment:
These sorts of comments are not necessary. TS function signature is already sufficient. Unless you are adding additional commentary to the options.